### DISCUSSION TITLE
Add blog docs

### DISCUSSION THREAD

**Discussion by antoinekm** [fuma-nama/fumadocs#1132] [Ideas]
Would be cool to have more docs about blog ? How it's working, search, pagination etc...

**Comment by Gitstar-OC** [comment]
#1113 refer to this and see the source code.

**Comment by fuma-nama** [comment]
I think it's a great idea to have a guide, kinda busy on other things right now but welcome for PRs!

**Reply by allentown521** [reply]
能否添加一个完整的代码示例，根据该指南遇到了问题：
类型“PageData & BaseCollectionEntry”上不存在属性“description”
import { buttonVariants } from '@/components/ui/button';

**Reply by allentown521** [reply]
并且不知道为啥文档和blog不能同时存在：
export const source = loader({
  baseUrl: '/docs',
  source: createMDXSource(docs, meta),
  i18n,
});

export const blog = loader({
  baseUrl: '/blog',
  source: createMDXSource(blogPosts, []),
});

增加了export const blog后，文档和blog都无法访问：
![image](https://github.com/user-attachments/assets/1a42c79e-3ca3-46d1-b3e9-f6f3777879e6)

blogPosts是按照指南写的
import { defineCollections, defineDocs, frontmatterSchema } from 'fumadocs-mdx/config';
import { z } from 'zod';

export const { docs, meta } = defineDocs();
export const blogPosts = defineCollections({
    type: 'doc',
    dir: 'content/blog',
    // add required frontmatter properties
    schema: frontmatterSchema.extend({
      author: z.string(),
      date: z.string().date().or(z.date()),
    }),
  });

**Reply by fuma-nama** [reply]
Hmm I guess `blogPosts` isn’t generated correctly, do you sure when restarting the dev server, the correct types are generated in `.source/index.d.ts`? (Which should include types for blogPosts)

**Reply by allentown521** [reply]
有的：
![image](https://github.com/user-attachments/assets/850779fb-ca4c-443d-8bf3-dd9c1a7d8902)
我可以提供最小化的项目源代码，能否帮忙看一下？

**Reply by allentown521** [reply]
[blog.zip](https://github.com/user-attachments/files/18177350/blog.zip)
不胜感激

**Reply by fuma-nama** [reply]
You need to import `blogPosts` from `.source` folder instead of the config file.

```ts
import { createMDXSource } from 'fumadocs-mdx';
import { loader } from 'fumadocs-core/source';
import { i18n } from '@/lib/i18n';
import { blogPosts, docs, meta } from '@/.source';

export const source = loader({
  baseUrl: '/docs',
  source: createMDXSource(docs, meta),
  i18n,
});

export const blog = loader({
  baseUrl: '/blog',
  source: createMDXSource(blogPosts, []),
});
```

I forgot to mention that `@/components/ui/button` is from Shadcn UI, will correct that later, you can remove the usage of it

**Reply by fuma-nama** [reply]
updated: https://fumadocs-dev.vercel.app/blog/make-a-blog

**Comment by Gitstar-OC** [comment]
I will work on adding a blog guide then, in which section should I add it ?

**Comment by allentown521** [comment]
works fine ，ths

Fuma Nama ***@***.***> 于2024年12月18日周三 20:58写道：

**Comment by allentown521** [comment]
还有个小问题：博客国际化不对，文档是对的
访问中文网页，但是加载的还是英文版的mdx文件：
[image: image.png]
这个是我的项目结构，麻烦帮我看下是哪里的问题，谢谢
[image: image.png]

Focus Apps ***@***.***> 于2024年12月18日周三 22:30写道：

**Reply by fuma-nama** [reply]
It's because you didn't pass i18n config to the blog loader

**Comment by allentown521** [comment]
我确实少添加了这个参数设置，不过按照你的指导我已经添加了，也删除了.next .source 重新yarn
dev，但是问题依旧，依然加载的是英文的mdx，：[image: image.png]

需要我提供任何的信息？

Fuma Nama ***@***.***> 于2024年12月19日周四 04:04写道：

**Reply by fuma-nama** [reply]
When you interact with the loader, make sure to specify the locale too, like `blog.getPage([params.slug], locale);` and `blog.getPages(locale);`

**Comment by allentown521** [comment]
works，请问这里如何国际化？
[image: image.png]

Fuma Nama ***@***.***> 于2024年12月19日周四 09:28写道：

**Reply by fuma-nama** [reply]
Hmm couldn’t see the image, may you describe that? If you are referring to the English text in example code (e.g. the “Back” button), you can display the localised text with conditional values like `locale === ‘cn’? “Chinese” : “English”` in JSX nodes.

**Reply by allentown521** [reply]
这些代码可以修改的我都改好了，我指的是这里：
![image](https://github.com/user-attachments/assets/2b208ad2-d13c-4ecf-bb47-b587ec8f0be9)

**Reply by fuma-nama** [reply]
Oh you are right, `InlineToc` didn’t have i18n support for trigger text, will add that soon

**Comment by allentown521** [comment]
Ok, thanks.

Fuma Nama ***@***.***> 于2024年12月19日周四 10:12写道：

**Reply by fuma-nama** [reply]
it is now supported to pass children to `<InlineTOC />` to change the trigger text

**Comment by allentown521** [comment]
Glad to hear it!

Fuma Nama ***@***.***> 于2024年12月19日周四 18:04写道：
